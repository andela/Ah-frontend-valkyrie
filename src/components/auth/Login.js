import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { login } from "../../actions/loginActions";
import "./styles/Login.css";
import LoginForm from "./login/LoginForm";
import { changelistener } from "../../../utils/globals";
import SocialAuth from "./social/SocialAuth";

export class Login extends Component {
  constructor(props) {
    super(props);
    this.state = {
      email: "",
      password: "",
    };
  }

  onSubmit = (event) => {
    event.preventDefault();
    const { email, password } = this.state;
    this.props.login(email, password);
  }


  render() {
    const { email, password } = this.state;
    const { isLoginPending, loginError } = this.props;
    return (
      <Container>
        <LoginForm
          onSubmit={this.onSubmit}
          email={email}
          password={password}
          changed={changelistener(this)}
          isLoginPending={isLoginPending}
          loginError={loginError}
        />
        <h4 className="text-center">Sign in with</h4>
        <SocialAuth />
        <ResetPasswordFunction />
      </Container>
    );
  }
}

export const Container = ({ children }) => (
  <div className="container">
    <div className="row Login">
      <div className="col-md-12 m-auto">
        <h4 className="text-center signIn">Sign in to your account</h4>
        {children}
      </div>
    </div>
  </div>
);

export const ResetPasswordFunction = () => (
  <div className="row mt-2">
    <div className="col-md-6">
      <p className="text-center mt-2">
        No accout?
        <a href="/users/register"> Sign up</a>
      </p>
    </div>
    <div className="col-md-6">
      <p className="text-center mt-2">
        Forgot password?
        {" "}
        <a href="/reset_password">Reset</a>
      </p>
    </div>
  </div>
);

Container.propTypes = {
  children: PropTypes.instanceOf(Array).isRequired,
};

Login.propTypes = ({
  isLoginPending: PropTypes.bool.isRequired,
  loginError: PropTypes.string,
  login: PropTypes.func.isRequired,
});

Login.defaultProps = {
  loginError: "",
};

const mapStateToProps = state => ({
  isLoginPending: state.loginReducer.isLoginPending,
  loginSuccess: state.loginReducer.loginSuccess,
  loginError: state.loginReducer.loginError,
  isAuthenticated: state.loginReducer.isAuthenticated,
});

const mapDispatchToProps = dispatch => ({
  login: (email, password) => dispatch(login(email, password)),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(Login);
