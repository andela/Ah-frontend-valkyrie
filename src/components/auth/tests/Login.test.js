import React from 'react';
import expect from 'expect';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import Login from '../Login';

const storeFake = ( state, action ) => ( {
  subscribe: jest.fn(),
  default: jest.fn(),
  getState: () => state,
  dispatch: () => action,
} );

const store = storeFake( {
  loginReducer:
  {
    isLoginPending: false, loginSuccess: {}, loginError: '', isAuthenticated: false,
  },
} );

describe( 'Login functionality', () => {
  const props = { isLoginPending: false, loginError: '' };
  const loginWrapper = mount(
    <Provider store={ store }>
      <Login { ...props } />
    </Provider>,
  );
  it( 'Login should render correctly', () => {
    const component = loginWrapper;
    expect( component ).toMatchSnapshot();
  } );

  it( 'handles the state changes on the Login Form', () => {
    const emailField = loginWrapper.find( '.form-control' ).first();
    emailField.value = 'testuser@app.com';
    const event = { target: { name: 'testName', value: 'testValue' } };
    emailField.simulate( 'change', event );
    expect( emailField.value ).toEqual( 'testuser@app.com' );
  } );

  it( 'handles submit', () => {
    const form = loginWrapper.find( '#login-form' );
    form.simulate( 'submit' );
    const { protocol, host } = window.location;
    expect( window.location.href ).toEqual( `${ protocol }//${ host }/` );
  } );
} );
