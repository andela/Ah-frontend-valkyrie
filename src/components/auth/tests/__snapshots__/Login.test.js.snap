// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login functionality Login should render correctly 1`] = `ReactWrapper {}`;
